functions.Rmd

```diff
@@ -16,26 +16,36 @@ permutation()
 
 ## Long lines
 
-If a function definition runs over multiple lines, indent the second line to 
-where the definition starts.
+If the function name and definition can't fit on a single line, place each argument on its own double indented line. The trailing `)` and leading `{` should go on the same line as the last argument.
 
 ```{r, eval = FALSE}
 # Good
+long_function_name <- function(
+    a = "a long argument",
+    b = "another argument",
+    c = "another long argument") {
+  # As usual code is indented by two spaces.
+}
+
+# No longer recommended because it doesn't work very well when the function
+# name is very long, which is not uncommon with S3 methods.
 long_function_name <- function(a = "a long argument",
```

Comment:
to clarify, this PR is actually a "breaking change" right?

i.e., instead of having two acceptable styles for multi-line function signatures, this PR aims to align behind the double-indented version, with the all-aligned "old" version now "deprecated"?

Reply:
Yes, except that we don't vigorously enforce the style guide, even internally. IOTW I see the style guide as a set of suggestions so nothing breaks even when we make bigger changes.